x-pack/plugins/security_solution/public/overview/components/detection_response/recent_cases_table/use_recent_cases_items.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { useState, useEffect, useCallback } from 'react';
+
+import { CaseStatuses } from '@kbn/cases-plugin/common';
+import { AllCases } from '@kbn/cases-plugin/common/ui';
+
+import { APP_ID } from '../../../../../common/constants';
+import { useGlobalTime } from '../../../../common/containers/use_global_time';
+import { useKibana } from '../../../../common/lib/kibana';
+import { addError } from '../../../../common/store/app/actions';
+
+export interface RecentCaseItem {
+  name: string;
+  note: string;
+  createdAt: string;
+  createdBy: string;
+  status: CaseStatuses;
+  id: string;
+}
+
+export interface UseRecentlyOpenedCasesProps {
+  skip: boolean;
+}
+
+export type UseRecentlyOpenedCases = (props: UseRecentlyOpenedCasesProps) => {
+  items: RecentCaseItem[];
+  isLoading: boolean;
+  updatedAt: number;
+};
+
+export const useRecentlyOpenedCases: UseRecentlyOpenedCases = ({ skip }) => {
+  const {
+    services: { cases },
+  } = useKibana();
+  const { to, from } = useGlobalTime();
+  const [isLoading, setIsLoading] = useState(true);
+  const [updatedAt, setUpdatedAt] = useState(Date.now());
+  const [items, setItems] = useState<RecentCaseItem[]>([]);
+
+  // This is a known issue of cases api, it doesn't accept date time format atm
+  // Once they fix this problem we can remove this two lines
+  // [TO-DO] #130979
+  const fromDate = moment(from).format('YYYY-MM-DD');
+  const toDate = moment(to).format('YYYY-MM-DD');
+
+  const fetchCases = useCallback(async () => {
+    try {
+      const casesResponse = await cases.api.cases.find({
+        from: fromDate,
+        to: toDate,
+        owner: APP_ID,
+        sortField: 'create_at',
+        sortOrder: 'desc',
+        page: 1,
+        perPage: 4,
+      });
+
+      setItems(parseRecentCases(casesResponse as unknown as AllCases));
+    } catch (err) {
+      addError(err);
+    }
+
+    setIsLoading(false);
+    setUpdatedAt(Date.now());
+  }, [cases.api.cases, fromDate, toDate]);
+
+  useEffect(() => {
+    if (skip) {
+      setIsLoading(false);
+    } else {
+      fetchCases();
+    }
+  }, [fetchCases, skip]);
+
+  return { items, isLoading, updatedAt };
+};
+
+function parseRecentCases(casesResponse: AllCases): RecentCaseItem[] {
+  const allCases = casesResponse.cases || [];
+
+  return allCases.reduce<RecentCaseItem[]>((accumalatedCases, currentCase) => {
+    accumalatedCases.push({
+      id: currentCase.id,
+      name: currentCase.title,
+      note: currentCase.description,
+      // @ts-ignore [TO-DO]
```

Comment:
wasnt sure if these were worth making a fuss about. but technically the linter wants camelCase so I Either tsignore, or ask cases to create the uniformity 🤔

Reply:
According to x-pack/plugins/cases/public/types.ts line 79
find cases api's type should be ```(query: CasesFindRequest) => Promise<CasesResponse>;```, 
but it returns the format below instead.
```
{
  "page": 1,
  "per_page": 4,
  "total": 3,
  "cases": [
    {
      "id": "c1f50550-c562-11ec-95ec-bb25c9b954e6",
      "version": "WzY4ODIsMV0=",
      "comments": [],
      "totalComment": 0,
      "totalAlerts": 0,
      "title": "case 3",
      "tags": [
        "case"
      ],
      "description": "case",
      "settings": {
        "syncAlerts": true
      },
      "owner": "securitySolution",
      "closed_at": null,
      "closed_by": null,
      "created_at": "2022-04-26T13:14:09.320Z",
      "created_by": {
        "username": "elastic",
        "email": null,
        "full_name": null
      },
      "status": "open",
      "updated_at": null,
      "updated_by": null,
      "connector": {
        "id": "none",
        "name": "none",
        "type": ".none",
        "fields": null
      },
      "external_service": null
    },
    {
      "id": "b9cfe840-c562-11ec-95ec-bb25c9b954e6",
      "version": "WzY4ODUsMV0=",
      "comments": [],
      "totalComment": 0,
      "totalAlerts": 0,
      "title": "case 2",
      "tags": [
        "case"
      ],
      "description": "case",
      "settings": {
        "syncAlerts": true
      },
      "owner": "securitySolution",
      "closed_at": null,
      "closed_by": null,
      "created_at": "2022-04-26T13:13:55.654Z",
      "created_by": {
        "username": "elastic",
        "email": null,
        "full_name": null
      },
      "status": "in-progress",
      "updated_at": "2022-04-26T13:14:19.381Z",
      "updated_by": {
        "full_name": null,
        "email": null,
        "username": "elastic"
      },
      "connector": {
        "id": "none",
        "name": "none",
        "type": ".none",
        "fields": null
      },
      "external_service": null
    },
    {
      "id": "1aeed790-c562-11ec-95ec-bb25c9b954e6",
      "version": "WzY4NzksMV0=",
      "comments": [],
      "totalComment": 0,
      "totalAlerts": 0,
      "title": "case",
      "tags": [
        "case"
      ],
      "description": "case",
      "settings": {
        "syncAlerts": true
      },
      "owner": "securitySolution",
      "closed_at": null,
      "closed_by": null,
      "created_at": "2022-04-26T13:09:29.103Z",
      "created_by": {
        "username": "elastic",
        "email": null,
        "full_name": null
      },
      "status": "open",
      "updated_at": null,
      "updated_by": null,
      "connector": {
        "id": "none",
        "name": "none",
        "type": ".none",
        "fields": null
      },
      "external_service": null
    }
  ],
  "count_open_cases": 2,
  "count_in_progress_cases": 1,
  "count_closed_cases": 0
}
```

Reply:
@jamster10 @angorayc The PR with the fix https://github.com/elastic/kibana/pull/131407. Do you want me to wait for this PR to get merged?